src/cursor/mod.rs

```diff
@@ -105,18 +107,32 @@ where
 
         Self {
             client: client.clone(),
-            wrapped_cursor: ImplicitSessionCursor::new(
+            wrapped_cursor: Some(ImplicitSessionCursor::new(
                 client,
                 spec,
                 PinnedConnection::new(pin),
                 provider,
-            ),
+            )),
             #[cfg(test)]
             kill_watcher: None,
             _phantom: Default::default(),
         }
     }
 
+    /// Update the type streamed values will be parsed as.
+    pub fn with_type<D>(mut self) -> Cursor<D>
+    where
+        D: DeserializeOwned + Unpin + Send + Sync,
+    {
+        Cursor {
```

Comment:
Unfortunately, it looks like `transmute` wouldn't be valid here - there's no guarantee that the monomorphized fields of `Cursor<T>` are laid out in the same order as the ones for `Cursor<D>`.  We could add `#[repr(C)]` to force that, but introducing that and `unsafe` doesn't seem worth saving an `unwrap`, especially since this is unlikely to be a performance bottleneck.